modules/portal/portal-cluster/src/main/java/com/liferay/portal/cluster/internal/ClusterExecutorImpl.java

```diff
@@ -408,12 +425,22 @@ protected Serializable handleReceivedClusterRequest(
 		Serializable payload = clusterRequest.getPayload();
 
 		if (payload instanceof ClusterNodeStatus) {
-			if (_memberJoined((ClusterNodeStatus)payload)) {
-				return ClusterRequest.createMulticastRequest(
-					_localClusterNodeStatus, true);
+			ClusterNodeStatus clusterNodeStatus = (ClusterNodeStatus)payload;
+
+			_memberJoined(clusterNodeStatus);
+
+			if (clusterRequest.isFireAndForget()) {
+				return null;
 			}
 
-			return null;
+			ClusterRequest notifyClusterRequest =
```

Comment:
This does not make sense to me.

This method is called handleReceivedClusterRequest(), it supposes to handle the incoming request then return a ClusterNodeResponse, regardless what actual content it has inside.

Here you are returning another ClusterRequest with local ClusterNodeStatus as payload. Won't it cause an infinite loop? On receiving it on your local host, it sees it as ClusterRequest, payload type as ClusterNodeStatus, it will just keep doing the same thing over and over, right?

Plus there is a real unit test failure in this pull.

Please fix them and resend.